### DISCUSSION TITLE
Support for B3

### DISCUSSION THREAD

**Discussion by zh17home** [WillCodeForCats/solaredge-modbus-multi#233] [General]
Hello, 

It will be possible to add also the third battery ?

Only see B1 and B2, B3 is not displayed.

Thanks,

**Comment by WillCodeForCats** [comment]
No, the modbus interface only defines 2 batteries.

**Comment by WillCodeForCats** [comment]
Please try loading this manually and see if it reports a battery 3 based on the existing pattern:

https://github.com/WillCodeForCats/solaredge-modbus-multi/archive/refs/heads/battery-three-test.zip

**Reply by cralwalker** [reply]
It does not report a battery 3.

**Reply by WillCodeForCats** [reply]
@cralwalker Please try [Release v2.4.9-pre.2](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.4.9-pre.2) if you have a third battery.

**Comment by Thebasestar** [comment]
Hi @WillCodeForCats , offset for 3rd battery is now available if you like to add to this integration.
I was able to add to my homeassistant editing the intergration of @binsetsu, but i think you can easy add to your as well.
find the details here:
https://github.com/binsentsu/home-assistant-solaredge-modbus/issues/149

Base

**Reply by WillCodeForCats** [reply]
Would you be able to test it?

**Reply by WillCodeForCats** [reply]
If you can test it please try [Release v2.4.9-pre.2](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.4.9-pre.2)

**Reply by Thebasestar** [reply]
I've just try it, it works!
![2024-01-20](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/95683280/9298274e-2073-4aa9-9f01-e26676a42b8d)

**Reply by WillCodeForCats** [reply]
Do all the entity and device values look good/different for the battery (I see they both say 11% charged so hoping that's not a duplicate)?

What firmware version is your inverter?

**Reply by Thebasestar** [reply]
Values are correct, all different. My inverter still on 4.17.218 Dsp1 1.100.1423  Dsp2 2.100.1302 soon i will update by myself if my installer don't 
![2024-01-20 (3)](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/95683280/75115631-c920-4264-8f9d-e78b3d021a03)
![2024-01-20 (2)](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/95683280/552088bb-8f6c-437d-9b6f-6f3a468e2d28)
![2024-01-20 (1)](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/95683280/6beff718-fd18-433a-ac7a-1b3a962b85d3)

**Reply by WillCodeForCats** [reply]
Thanks for testing this, it looks good.

**Comment by WillCodeForCats** [comment]
Add support for third battery in PR #508

Please try [Release v2.4.9-pre.2](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.4.9-pre.2) if you have a third battery.

**Reply by WillCodeForCats** [reply]
Please report your firmware version too if it works.

**Reply by saurabh984** [reply]
SE6000H-RW000BEN4
Inverter Firmware: 0004.0018.0032
Battery Firmware: SMCU 1.1.9 DCDC 0.0.81 BMS 0.6.9

Third battery is visible but none of them are reporting Available Energy. It's `unknown` for all three

**Comment by zh17home** [comment]
Thanks WillCode!

![image](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/65257232/b4ac2e1a-30b5-43af-82a7-524dc9c04fd9)

**Reply by WillCodeForCats** [reply]
Looks good, thanks for the report!

**Comment by Thebasestar** [comment]
Updated my inverter to latest version yesterday and 3battery integration still work.

**Reply by cyberkryten** [reply]
I'd advise caution on updating.  

SE remotely updated mine on Friday to the latest version, along with updating Battery Firmware in an attempt to try to solve other issues, but some of the updates failed.  

I now have no batteries or Backup Gateway visible to the inverter.  May be totally unrelated, but just thought I'd caution for others in case this latest release has issues

**Reply by Thebasestar** [reply]
Which version do you have now?
On mine after the update
CPU 4.19.39 DSP1 1.100.1631 DSP2 2.100.1412
Meter 1081
But batteries failed the update 3 times and are still on
SMCU 1.0.67 DCDC 0.0.79 BMS 0.6.5
With this update the maximum power to charge/discharge the batteries is now set to 7000w, before was 5000w so for me better, i can now use full power (before i had to put on remote and set to 6000w to use full power on batteries).
Anyway thanks for the advise

**Reply by cyberkryten** [reply]
Ah, that's not the very latest inverter firmware - 4.19.39 is from December.

Hard to know what the state actually is now as the batteries and gateway failed to come back online properly and have been offline totally since the inverter reset the night after the update.  I've already had one battery replaced as it was DOA!

Two batteries have solid red fault lights and one has flashing blue comms light.  The Active Power Limit seems to be fixed at 36% and the inverter won't produce more than 3.6kW AC (so was clipping all weekend).

Inverter is now 
`cpuVersion: 4.20.32 `
I don't see DSP versions reported anywhere on either 

Batteries last reports were the following - it's unclear at what point things failed and whether newer versions got loaded before comms was lost:

```
firmwareVersion: SMCU 1.1.29 DCDC 1.0.28 BMS 1.0.51
firmwareVersion: SMCU 1.1.35 DCDC 1.0.28 BMS 1.0.51
firmwareVersion: SMCU 1.0.67 DCDC 1.0.26 BMS 1.1.6
```
So the three batteries have a very eclectic mix of versions!

https://www.solaredge.com/en/support/setapp-inverters-firmware
Looking at the website, latest versions are:
```
SolarEdge Home Battery 400V SMCU    1.1.35
SolarEdge Home Battery 400V DCDC    0.1.85 / 1.1.32
SolarEdge Home Battery 400V BMS       0.7.0 / 1.1.6
```

Obviously, this may all be totally unrelated to the firmware update, but the inverter doing a reset 9 hours after SE reported that 'auto update has not been successful' and the whole system going fubar is a bit of a coincidence!

**Reply by Thebasestar** [reply]
Well i did the update on feb, before the latest version...
You had a bad update caused by them so they should resolve for free, there is an usb port inside the inverter and they can use it also for downgrade . You can also try with setapp yourself, but maybe better wait the service...

**Reply by cyberkryten** [reply]
I'd be interested to know if anyone else is on 4.20.32 and using cabled connections to Backup Gateway and Batteries (though I wouldn't suggest upgrading if you haven't, just in case).

They're now saying they need to fit their new wireless connection board to the inverter and switch over to that for battery comms instead of using the cables (which makes no sense, as I'd always use cabled connections over wireless for reliability).  That makes me suspect they may be aware of issues. 

Also, does anyone have batteries (I have three) connected using the Solar Edge wireless comms are are you able to read data about them via Modbus TCP using this integration?  

It isn't entirely clear if the Inverter will still expose the battery information read from the wireless connection over the TCP Modbus interface.  If not, then that really isn't going to be a viable solution for me.

**Reply by cralwalker** [reply]
I do.

Who are 'they' and who pay for the kit and installation?

Do you have a reference to share?

**Reply by cyberkryten** [reply]
"They" are SolarEdge Tech via my installer.  Still waiting for the new board to be delivered and installed, at their cost, so will see when that gets fitted.
